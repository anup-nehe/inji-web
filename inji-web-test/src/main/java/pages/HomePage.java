package pages;


import base.BasePage;
import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.openqa.selenium.TimeoutException;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.interactions.Actions;

import java.util.Arrays;
import java.util.HashSet;
import java.util.List;


public class HomePage extends BasePage {

    private WebDriver driver;


    public HomePage(WebDriver driver) {
        this.driver = driver;
    }

    public void clickOnHelp() {
        if (isElementIsVisible(driver, By.xpath("//div[@data-testid='Header-Menu-Help']"))) {
            clickOnElement(driver, By.xpath("//div[@data-testid='Header-Menu-Help']"));
        } else {
            clickOnElement(driver, By.xpath("//li[@data-testid='Header-Menu-Help']"));
        }
    }

    public boolean isHelpPageDisplayed() {
        return isElementIsVisible(driver, By.xpath("//div[@data-testid='Header-Menu-Help']"));
    }

    public boolean isHeaderContanerDisplayed() {
        return isElementIsVisible(driver, By.xpath("//div[@data-testid='Header-Container']"));
    }

    public boolean isFooterIsDisplayedOnHomePage() {
        return isElementIsVisible(driver, By.xpath("//*[@data-testid='Footer-Text']"));
    }

    public String getFooterText() {
        return getElementText(driver, By.xpath("//*[@data-testid='Footer-Text']"));
    }

    public boolean isHomePageContainerDisplayed() {
        return isElementIsVisible(driver, By.xpath("//*[@data-testid='Home-Page-Container']"));
    }

    public boolean isAboutDisplayed() {
        return isElementIsVisible(driver, By.xpath("//*[@data-testid='Header-Menu-AboutInji']"));
    }

    public boolean isSerchBoxDisplayed() {
        return isElementIsVisible(driver, By.xpath("//*[@data-testid='Search-Issuer-Input']"));
    }

    public boolean isIssuerLogoDisplayed() {
        return isElementIsVisible(driver, By.xpath("//*[@data-testid='ItemBox-Logo']"));
    }

    public void clickOnAboutInji() {
        clickOnElement(driver, By.xpath("//*[@data-testid='Header-Menu-AboutInji']"));
    }


    public Boolean isLogoDisplayed() {
        try {
            Thread.sleep(3000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        return isElementIsVisible(driver, By.xpath("//div[@data-testid='Header-InjiWeb-Logo-Container']"));
    }

    public String isPageTitleDisplayed() {
        return driver.getCurrentUrl();
    }

    public String isHomePageTextDisplayed() {
        return getElementText(driver, By.xpath("//h2[@data-testid='IntroBox-Text']"));
    }

    public String isHomePageDescriptionTextDisplayed() {
        return getElementText(driver, By.xpath("//p[@data-testid='IntroBox-SubText']"));
    }

    public void clickOnDownloadMosipCredentials() {
        clickOnElement(driver, By.xpath("(//h3[@data-testid='ItemBox-Text'])[1]"));
    }

    public Boolean isListOfCredentialsTypeDisplayed() {
        try {
            Thread.sleep(3000);
        } catch (InterruptedException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
        return isElementIsVisible(driver, By.xpath("//div[@data-testid='HeaderTile-Text']"));
    }

    public void clickOnVerify() {
        clickOnElement(driver, By.xpath("//button[@id='verify_otp']"));
        try {
            Thread.sleep(3000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        if (isElementIsVisible(driver, By.xpath("//button[@id='verify_otp']"))) {
            clickOnElement(driver, By.xpath("//button[@id='verify_otp']"));
        }
    }

    public Boolean isSuccessMessageDisplayed() {
        try {
            Thread.sleep(6000);
            ;
        } catch (InterruptedException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
        return isElementIsVisible(driver, By.xpath("//*[@data-testid='DownloadResult-Title']"));
    }

    public Boolean isMosipNationalIdDisplayed() {
        try {
            Thread.sleep(2000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        return isElementIsVisible(driver, By.xpath("//h3[@data-testid='ItemBox-Text']"));
    }

    //
    public void enterIssuersInSearchBox(String string) {
        enterText(driver, By.xpath("//input[@type='text']"), string);
        if (isElementIsVisible(driver, By.xpath("//p[@data-testid='IntroBox-SubText']"))) {
            clickOnElement(driver, By.xpath("//p[@data-testid='IntroBox-SubText']"));
        }
    }

    //
    public Boolean isGoHomeButtonDisplayed() {
        try {
            Thread.sleep(4000);
        } catch (InterruptedException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
        return isElementIsVisible(driver, By.xpath("//button[text()='Go To Home']"));
    }

    public Boolean isBackButtonDisplayed() {
        return isElementIsVisible(driver, By.xpath("//*[@class='cursor-pointer']"));
    }

    //
    public Boolean isLanguageDisplayed() {
        return isElementIsVisible(driver, By.xpath("//button[@class='inline-flex items-center']"));
    }

    //
    public void clickOnLanguageButton() {
        try {
            Thread.sleep(2000);
        } catch (InterruptedException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
        if (isElementIsVisible(driver, By.xpath("//div[@data-testid='HeaderTile-Text']"))) {
            clickOnElement(driver, By.xpath("//div[@data-testid='HeaderTile-Text']"));
        }
        clickOnElement(driver, By.xpath("//*[@data-testid='Language-Selector-Button']"));

    }

    public boolean verifyLanguagesInLanguageFilter() {
        List<String> expectedLanguages = Arrays.asList("English", "தமிழ்", "ಕನ್ನಡ", "हिंदी", "Français", "عربي");
        List<String> actualLanguages = null;
        try {
            actualLanguages = getElementTexts(driver, By.xpath("//ul[@class='py-1 divide-y divide-gray-200']//li"));

        } catch (TimeoutException e) {
            e.printStackTrace();
        }
        System.out.println("expectedLanguages" + expectedLanguages);
        System.out.println("actualLanguages" + actualLanguages);
        return new HashSet<>(expectedLanguages).equals(new HashSet<>(actualLanguages));
    }

    public void selectOtherLangauge() {
        clickOnElement(driver, By.xpath("(//button[@type='button'])[3]"));

        try {
            Thread.sleep(7000);
        } catch (InterruptedException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
    }

    //
    public Boolean isNoIssuerFoundMessageDisplayed() {
        try {
            Thread.sleep(2000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        return isElementIsVisible(driver, By.xpath("//p[@data-testid='EmptyList-Text']"));
    }
//
//	public void isPdfDownloaded() {
//		 Download download = page.waitForDownload((
//	}

    public void clickOnArabicLanguage() {
        clickOnElement(driver, By.xpath("//button[contains(text(), 'عربي')]"));
    }

    public void clickOnTamilLanguage() {
        clickOnElement(driver, By.xpath("//button[contains(text(), 'தமிழ்')]"));
    }

    public void clickOnKannadaLanguage() {
        clickOnElement(driver, By.xpath("//button[contains(text(), 'ಕನ್ನಡ')]"));
    }

    public void clickOnHindiLanguage() {
        clickOnElement(driver, By.xpath("//button[contains(text(), 'हिंदी')]"));
    }

    public void clickOnFranchLanguage() {
        clickOnElement(driver, By.xpath("//button[contains(text(), 'Français')]"));
    }

    public String isCredentialTypesDisplayed() {
        return getElementText(driver, By.xpath("//div[@data-testid='HeaderTile-Text']"));
    }

    public String isNationalIdentityDepartmentTextDisplayed() {
        return getElementText(driver, By.xpath("//span[@data-testid='NavBar-Text']"));
    }

    public String isVeridoniaInsuranceCompanyTextDisplayed() {
        return getElementText(driver, By.xpath("//span[@data-testid='NavBar-Text']"));
    }

    public static void scrollDownByPage(WebDriver driver) {
        Actions actions = new Actions(driver);
        actions.sendKeys(Keys.PAGE_DOWN).build().perform();
    }

    public void ClickOnHelpForMobileBrowse() {
        if (isElementIsVisible(driver, By.xpath("//div[@data-testid='Header-InjiWeb-Logo-Container']/div"))) {
            clickOnElement(driver, By.xpath("//div[@data-testid='Header-InjiWeb-Logo-Container']/div"));
        }
    }

    public void verifyInvaildMessage() {
        isElementIsVisible(driver, By.xpath("//p[text()='Invalid Session']"));
    }


    public String isListOfIssuersTextDisplayed() {
        return getElementText(driver, By.xpath("(//*[@data-testid='HeaderTile-Text'])[1]"));
    }

    public String isListOfIssuersDescriptionTextDisplayed() {
        return getElementText(driver, By.xpath("(//*[@data-testid='HeaderTile-Text'])[2]"));
    }

    public boolean isAboutPageDisplayed() {
        return isElementIsVisible(driver, By.xpath("//p[text()='Overview']"));
    }
}
