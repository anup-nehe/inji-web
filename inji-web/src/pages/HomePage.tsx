import React from "react";
import {useNavigate} from "react-router-dom";
import {HomeBanner} from "../components/Home/HomeBanner";
import {HomeFeatures} from "../components/Home/HomeFeatures";
import {HomeQuickTip} from "../components/Home/HomeQuickTip";
import {toast} from "react-toastify";
import {useTranslation} from "react-i18next";

export const HomePage:React.FC = () => {
    const navigate = useNavigate();
    const {t} = useTranslation("HomePage");
    return <div className={"pb-20 flex flex-col gap-y-4 "}>
        <HomeBanner onClick={() => navigate("/issuers")} />
        <HomeFeatures/>
        <HomeQuickTip  onClick={() => toast.warning(t("QuickTip.toastText"))} />
    </div>
}
