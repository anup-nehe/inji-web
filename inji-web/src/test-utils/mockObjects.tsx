import { IssuerWellknownObject } from '../types/data';
import { LandingPageWrapperProps } from '../components/Common/LandingPageWrapper';
import { NavBarProps } from '../types/components';
export const mockDisplayArrayObject = {
    name: "Name",
    language: "en",
    locale: "en",
    logo: {
        url: "https://url.com",
        alt_text: "alt text of the url"
    },
    title: "Title",
    description: "Description",
};

export const mockCredentials: IssuerWellknownObject = {
    credential_issuer: "",
    credential_endpoint: "",
    authorization_servers: [""],
    credential_configurations_supported: {
        InsuranceCredential: {
            format: "ldp_vc",
            scope: "mosip_ldp_vc",
            order: [],
            display: [{
                name: "Name1",
                language: "en",
                locale: "en",
                logo: {
                    url: "https://url.com",
                    alt_text: "alt text of the url"
                },
                title: "Title",
                description: "Description",
            }],
            proof_types_supported: [],
            credential_definition: {
                type: [],
                credentialSubject: {
                    fullName: {
                        display: [{
                            name: "Name1",
                            language: "en",
                            locale: "en",
                            logo: {
                                url: "https://url.com",
                                alt_text: "alt text of the url"
                            },
                            title: "Title",
                            description: "Description",
                        }],
                    }
                }
            }
        },
        AnotherCredential: {
            format: "ldp_vc",
            scope: "mosip_ldp_vc",
            order: [],
            display: [{
                name: "Name2",
                language: "en",
                locale: "en",
                logo: {
                    url: "https://url.com",
                    alt_text: "alt text of the url"
                },
                title: "Title",
                description: "Description",
            }],
            proof_types_supported: [],
            credential_definition: {
                type: [],
                credentialSubject: {
                    fullName: {
                        display: [{
                            name: "Name2",
                            language: "en",
                            locale: "en",
                            logo: {
                                url: "https://url.com",
                                alt_text: "alt text of the url"
                            },
                            title: "Title",
                            description: "Description",
                        }],
                    }
                }
            }
        }
    }
};

export const mockSearchCredential ={
    credential_configurations_supported: {
        InsuranceCredential: {
            display: [{
                name: "Insurance Credential",
                language: "en",
                locale: "en",
                logo: {
                    url: "https://url.com",
                    alt_text: "alt text of the url"
                },
                title: "Title",
                description: "Description",
            }],
        },
        AnotherCredential: {
            display: [{
                name: "Another Credential",
                language: "en",
                locale: "en",
                logo: {
                    url: "https://url.com",
                    alt_text: "alt text of the url"
                },
                title: "Title",
                description: "Description",
            }],
        }
    }
};

export const mockIssuerObject = {
        name: 'Test Issuer',
        desc: 'Test Description',
        protocol: 'OTP' as 'OTP', // Explicitly set the type to 'OTP' or 'OpenId4VCI'
        issuer_id: 'test-issuer',
        authorization_endpoint: 'https://auth.test.com',
        credentials_endpoint: 'https://credentials.test.com',
        display: [{
            name: "Name",
            language: "en",
            locale: "en",
            logo: {
                url: "https://url.com",
                alt_text: "alt text of the url"
            },
            title: "Title",
            description: "Description",
        }],
        client_id: 'test-client-id',
        redirect_uri:'test-redirect-uri',
        token_endpoint:'test-token_endpoint',
        proxy_token_endpoint:'test-proxy_token_endpoint',
        client_alias:'',
        ovp_qr_enabled: true,
        scopes_supported: ['scope1', 'scope2']
}

export const mockIssuerObjectList =  [
    {
        name: 'Issuer 1',
        desc: 'Description 1',
        protocol: 'OpenId4VCI',
        issuer_id: 'issuer1',
        authorization_endpoint: 'https://issuer1.com/auth',
        credentials_endpoint: 'https://issuer1.com/credentials',
        display: [
            {
                name: 'Issuer 1',
                language: 'en',
                locale: 'en-US',
                logo: { url: 'https://issuer1.com/logo.png', alt_text: 'Issuer 1 Logo' },
                title: 'Issuer 1 Title',
                description: 'Issuer 1 Description',
            },
        ],
        client_id: 'client1',
        redirect_uri: 'https://issuer1.com/redirect',
        token_endpoint: 'https://issuer1.com/token',
        proxy_token_endpoint: 'https://issuer1.com/proxy-token',
        client_alias: 'client1-alias',
        ovp_qr_enabled: true,
        scopes_supported: ['openid', 'profile'],
    },
    {
        name: 'Issuer 2',
        desc: 'Description 2',
        protocol: 'OpenId4VCI',
        issuer_id: 'issuer2',
        authorization_endpoint: 'https://issuer2.com/auth',
        credentials_endpoint: 'https://issuer2.com/credentials',
        display: [
            {
                name: 'Issuer 2',
                language: 'en',
                locale: 'en-US',
                logo: { url: 'https://issuer2.com/logo.png', alt_text: 'Issuer 2 Logo' },
                title: 'Issuer 2 Title',
                description: 'Issuer 2 Description',
            },
        ],
        client_id: 'client2',
        redirect_uri: 'https://issuer2.com/redirect',
        token_endpoint: 'https://issuer2.com/token',
        proxy_token_endpoint: 'https://issuer2.com/proxy-token',
        client_alias: 'client2-alias',
        ovp_qr_enabled: false,
        scopes_supported: ['openid', 'profile'],
    },
]

export const mockLandingPageWrapperProps: LandingPageWrapperProps = {
    icon: <div data-testid="Test-Icon">Icon</div>,
    title: "Test Title",
    subTitle: "Test SubTitle",
    gotoHome: true,
};

export const mockNavBarProps: NavBarProps = {
    title: 'Test Title',
    link: '/test-link',
    search: true,
};