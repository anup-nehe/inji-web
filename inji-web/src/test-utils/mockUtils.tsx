import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { reduxStore } from '../redux/reduxStore';

export const mockUseTranslation = () => {
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
            t: (key: string) => key,
        }),
    }));
};

export const mockUseNavigate = () => {
    const mockNavigate = jest.fn();
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useNavigate: mockNavigate,
    }));
};

export const mockUseSearchCredentials = () => {
    jest.mock('../components/Credentials/SearchCredential', () => ({
        SearchCredential: () => <></>,
    }));
};

export const mockUseSelector = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useSelector: jest.fn(),
    }));
};

export const setMockUseSelectorState = (state: any) => {
    const useSelectorMock = require('react-redux').useSelector;
    useSelectorMock.mockImplementation((selector: any) => selector(state));
};

export const mockUseGetObjectForCurrentLanguage = () => {
    jest.mock('../utils/i18n', () => ({
        getObjectForCurrentLanguage: jest.fn(),
    }));
};

export const wrapUnderRouter = (children: React.ReactNode) => {
    return <Router>{children}</Router>;
};

export const mockUseToast = () => {
    jest.mock('react-toastify', () => ({
        toast: jest.fn(),
        ToastContainer: jest.requireActual('react-toastify').ToastContainer,
    }));
};

export const mockUseDispatch = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useDispatch: jest.fn(),
    }));
};

export const mockUseSpinningLoader = () => {
    jest.mock('../components/Common/SpinningLoader', () => ({
        SpinningLoader: () => <></>,
    }));
};

export const mockUseLanguageSelector = () => {
    jest.mock('../components/Common/LanguageSelector', () => ({
        LanguageSelector: () => <></>,
    }));
};

interface RenderWithProviderOptions extends Omit<RenderOptions, 'queries'> {}

export const renderWithProvider = (element: ReactElement, options?: RenderWithProviderOptions) => {
    return render(
        <Provider store={reduxStore}>
            <Router>
                {element}
            </Router>
        </Provider>,
        options
    );
};

