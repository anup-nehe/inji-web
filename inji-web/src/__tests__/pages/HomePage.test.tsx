import React from 'react';
import { screen, waitFor, fireEvent } from '@testing-library/react';
import { HomePage } from '../../pages/HomePage';
import { reduxStore } from '../../redux/reduxStore';
import { storeIssuers } from '../../redux/reducers/issuersReducer';
import { renderWithRouter, mockusei18n,mockUseFetch,mockUseToast } from '../../test-utils/mockUtils';
import { mockIssuerObject } from '../../test-utils/mockObjects';

mockusei18n();
const mockUseFetchHook = jest.fn();
mockUseFetch();
mockUseToast();

//todo : extract the local method to mockUtils, which is added to bypass the routing problems
jest.mock('react-toastify', () => ({
  toast: {
    error: jest.fn(),
  },
}));

describe('Testing the Layout of HomePage', () => {
  test('Check if the layout is matching with the snapshots', () => {
    mockUseFetchHook.mockReturnValue({ state: 'DONE', fetchRequest: jest.fn() });
    const { asFragment } = renderWithRouter(<HomePage />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Testing the Functionality of HomePage', () => {
  beforeEach(() => {
    mockusei18n();
    reduxStore.dispatch(storeIssuers([mockIssuerObject]));
  });

  test('Check if IntroBox and SearchIssuer components are rendered', () => {
    mockUseFetchHook.mockReturnValue({ state: 'DONE', fetchRequest: jest.fn() });
    const { asFragment } = renderWithRouter(<HomePage />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Check if it displays error message if state is ERROR', async () => {
    mockUseFetchHook.mockReturnValue({ state: 'ERROR', fetchRequest: jest.fn() });
    renderWithRouter(<HomePage />);
    await waitFor(() => {
      expect(require('react-toastify').toast.error).toHaveBeenCalledWith('The service is currently unavailable now. Please try again later.');
    });
  });

  test('Check if search input filters issuers correctly', async () => {
    mockUseFetchHook.mockReturnValue({ state: 'DONE', fetchRequest: jest.fn() });
    renderWithRouter(<HomePage />);
    const searchInput = screen.getByTestId('Search-Issuer-Input');
    fireEvent.change(searchInput, { target: { value: 'Test' } });
    await waitFor(() => {
      expect(searchInput).toHaveValue('Test');
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
