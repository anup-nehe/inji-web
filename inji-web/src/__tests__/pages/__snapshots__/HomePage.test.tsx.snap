// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Functionality of HomePage Check if IntroBox and SearchIssuer components are rendered 1`] = `
<DocumentFragment>
  <div
    data-testid="Home-Page-Container"
  >
    <div
      class="container mx-auto mt-8 flex flex-col px-10"
    >
      <div
        class="mb-20"
      >
        <div
          class="text-center pt-20 pb-10"
          data-testid="IntroBox-Container"
        >
          <h2
            class="text-3xl text-iw-title "
            data-testid="IntroBox-Text"
          >
            Access your Verifiable Credentials with ease!
          </h2>
          <p
            class="mt-2 text-lg text-iw-subTitle"
            data-testid="IntroBox-SubText"
          >
            Securely download and share your credentials instantly.
          </p>
        </div>
        <div
          class="flex justify-center items-center w-full"
        >
          <div
            class="sm:w-8/12 w-full flex justify-start items-center bg-iw-background shadow-md shadow-iw-shadow"
            data-testid="Search-Issuer-Container"
          >
            <svg
              class="m-5"
              color="var(--iw-color-searchIcon)"
              data-testid="Search-Issuer-Search-Icon"
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
            <input
              class="py-6 w-11/12 text-iw-searchTitle focus:outline-none overflow-ellipsis mr-10"
              data-testid="Search-Issuer-Input"
              placeholder="Search Issuers by name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="flex justify-center items-center"
        data-testid="SpinningLoader-Container"
      >
        <div
          aria-busy="true"
          aria-label="oval-loading"
          class="sc-beySPh gZmPOF"
          data-testid="oval-loading"
          role="progressbar"
        >
          <svg
            data-testid="oval-svg"
            height="80"
            stroke="var(--iw-color-spinningLoaderPrimary)"
            viewBox="-20.5 -20.5 43 43"
            width="80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                data-testid="oval-secondary-group"
                stroke-width="3"
                transform="translate(1 1)"
              >
                <circle
                  cx="0"
                  cy="0"
                  r="20"
                  stroke="var(--iw-color-spinningLoaderSecondary)"
                  stroke-opacity=".5"
                  stroke-width="3"
                />
                <path
                  d="M20 0c0-9.94-8.06-20-20-20"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="1s"
                    from="0 0 0"
                    repeatCount="indefinite"
                    to="360 0 0"
                    type="rotate"
                  />
                </path>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing the Layout of HomePage Check if the layout is matching with the snapshots 1`] = `
<DocumentFragment>
  <div
    data-testid="Home-Page-Container"
  >
    <div
      class="container mx-auto mt-8 flex flex-col px-10"
    >
      <div
        class="mb-20"
      >
        <div
          class="text-center pt-20 pb-10"
          data-testid="IntroBox-Container"
        >
          <h2
            class="text-3xl text-iw-title "
            data-testid="IntroBox-Text"
          >
            Access your Verifiable Credentials with ease!
          </h2>
          <p
            class="mt-2 text-lg text-iw-subTitle"
            data-testid="IntroBox-SubText"
          >
            Securely download and share your credentials instantly.
          </p>
        </div>
        <div
          class="flex justify-center items-center w-full"
        >
          <div
            class="sm:w-8/12 w-full flex justify-start items-center bg-iw-background shadow-md shadow-iw-shadow"
            data-testid="Search-Issuer-Container"
          >
            <svg
              class="m-5"
              color="var(--iw-color-searchIcon)"
              data-testid="Search-Issuer-Search-Icon"
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
            <input
              class="py-6 w-11/12 text-iw-searchTitle focus:outline-none overflow-ellipsis mr-10"
              data-testid="Search-Issuer-Input"
              placeholder="Search Issuers by name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="flex justify-center items-center"
        data-testid="SpinningLoader-Container"
      >
        <div
          aria-busy="true"
          aria-label="oval-loading"
          class="sc-beySPh gZmPOF"
          data-testid="oval-loading"
          role="progressbar"
        >
          <svg
            data-testid="oval-svg"
            height="80"
            stroke="var(--iw-color-spinningLoaderPrimary)"
            viewBox="-20.5 -20.5 43 43"
            width="80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                data-testid="oval-secondary-group"
                stroke-width="3"
                transform="translate(1 1)"
              >
                <circle
                  cx="0"
                  cy="0"
                  r="20"
                  stroke="var(--iw-color-spinningLoaderSecondary)"
                  stroke-opacity=".5"
                  stroke-width="3"
                />
                <path
                  d="M20 0c0-9.94-8.06-20-20-20"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="1s"
                    from="0 0 0"
                    repeatCount="indefinite"
                    to="360 0 0"
                    type="rotate"
                  />
                </path>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
